framework/console/controllers/ServeController.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\console\controllers;
+
+use Yii;
+use yii\console\Controller;
+use yii\helpers\Console;
+
+/**
+ * Runs PHP built-in web server
+ *
+ * In order to access server from remote machines use 0.0.0.0:8000. That is especially useful when running server in
+ * a virtual machine.
+ *
+ * @author Alexander Makarov <sam@rmcreative.ru>
+ * @since 2.0.7
+ */
+class ServeController extends Controller
+{
+    const EXIT_CODE_NO_DOCUMENT_ROOT = 2;
+    const EXIT_CODE_NO_ROUTING_FILE = 3;
+    const EXIT_CODE_ADDRESS_TAKEN_BY_ANOTHER_SERVER = 4;
+    const EXIT_CODE_ADDRESS_TAKEN_BY_ANOTHER_PROCESS = 5;
+
+    /**
+     * @var int port to serve on. Either "host" or "host:port".
+     */
+    public $port = 8080;
+
+    /**
+     * @var string path to directory to serve
+     */
+    public $docroot = 'web';
+
+    /**
+     * @var string path to router script.
+     * See https://secure.php.net/manual/en/features.commandline.webserver.php
+     */
+    public $router;
+
+    /**
+     * Runs PHP built-in web server
+     *
+     * @param string $address address to serve on
+     *
+     * @return int
+     */
+    public function actionIndex($address = 'localhost')
```

Comment:
I think it's not a daemon background service that should have start, stop, restart